setup.py

```diff
@@ -27,6 +27,7 @@
     "fsspec >= 0.6.0",
     "toolz >= 0.8.2",
     "partd >= 0.3.10",
+    "importlib_metadata >= 3.6.0",
```

Comment:
Is this only necessary for older versions of Python? I think you can use [environment markers](https://www.python.org/dev/peps/pep-0508/#environment-markers) to ensure it's only required if necessary.

Reply:
It is updated ahead of the builtin `importlib.metadata`. It gets a bit awkward if we try to use the builtin, because the API seems to have changed a number of times, and I'd not want to pin this too strongly.

Reply:
Has it changed in ways that affect the way we use it? I'd prefer to keep our dependencies as minimal as possible. I'm willing to take on a bit of compatibility code if it means Python 3.8+ doesn't require a new dependency.

Reply:
See https://github.com/dask/dask/pull/7688#discussion_r652756468 for an example `import` compatibility snippet. @agoose77 is that sufficient, or are there other changes we need to account for?